File: data/llama3_8B_ins_results/readable_format_HPEC/cpp/file179.cpp
#include <complex>
#include <iostream>

void cffts2(int is, int d1, int d2, int d3, std::complex<double> x[d1+1][d2][d3], std::complex<double> xout[d1+1][d2][d3], std::complex<double> y1[fftblockpad][d2], std::complex<double> y2[fftblockpad][d2]) {
  int logd2 = std::log2(d2);
  int fftblock = 256; // assume fftblock is a constant

  if (timers_enabled) {
    timer_start(T_ffty);
  }

  for (int k = 0; k < d3; k++) {
    for (int in = 0; in < d1/fftblock; in++) {
      int ii = in * fftblock;
      for (int j = 0; j < d2; j++) {
        for (int i = 0; i < fftblock; i++) {
          y1[i][j] = x[i+ii][j][k];
        }
      }
    }
  }

  cfftz(is, logd2, d2, y1, y2);

  for (int j = 0; j < d2; j++) {
    for (int i = 0; i < fftblock; i++) {
      xout[i+ii][j][k] = y1[i][j];
    }
  }

  if (timers_enabled) {
    timer_stop(T_ffty);
  }
}

File: data/CodeLlama/readable_format_HPEC/fortran/file179.f90
subroutine cffts2(is, d1, d2, d3, x, xout, y1, y2)


use ft_data
implicit none

integer is, d1, d2, d3, logd2
double complex x(d1+1,d2,d3)
double complex xout(d1+1,d2,d3)
double complex y1(fftblockpad, d2), y2(fftblockpad, d2)
integer i, j, k, ii, in

logd2 = ilog2(d2)

if (timers_enabled) call timer_start(T_ffty)
do k = 1, d3
do in = 0, d1/fftblock - 1
ii = in*fftblock
do j = 1, d2
do i = 1, fftblock
y1(i,j) = x(i+ii,j,k)
enddo
enddo

call cfftz (is, logd2, d2, y1, y2)

do j = 1, d2
do i = 1, fftblock
xout(i+ii,j,k) = y1(i,j)
enddo
enddo
enddo
enddo
if (timers_enabled) call timer_stop(T_ffty)

return
end
